token-sdk/README.md

```diff
@@ -0,0 +1,339 @@
+-   Production configuration (e.g. deployment, networking, security, resilience, key management)
+
+## Getting started
+
+Prerequisites:
+
+- bash
+- golang 1.20+
+- git
+- docker
+- docker-compose
+
+### Install dependencies
+
+Download the Fabric docker images and binaries. The code only works with Fabric CA 1.5.7+, so even if you cloned the fabric-samples repo before, you may have to re-run it to get the latest versions.
+
+```bash 
+curl -sSLO https://raw.githubusercontent.com/hyperledger/fabric/main/scripts/install-fabric.sh && chmod +x install-fabric.sh
+./install-fabric.sh docker binary
+```
+
+Make sure that the new binaries are in your path. Change the following line (replace `<your/path/to/> with the actual path`) and add them to your `~/.bashrc` or `~/.zshrc` file (and restart your terminal):
+
+```bash
+export PATH=</your/path/to/>fabric-samples/bin:$PATH
+```
+
+> Note: if you are *not* running this code from the fabric-samples/token-sdk folder, set the following environment variable:
+> ```bash
+> export TEST_NETWORK_HOME=</your/path/to>/fabric-samples/test-network
+> ```
+
+Validate that the CA is at 1.5.7 by executing `fabric-ca-client version`.
+
+Install tokengen.
+
+```bash
+go install github.com/hyperledger-labs/fabric-token-sdk/cmd/tokengen@v0.3.0
+```
+
+Now go to the root folder of your token-sdk sample, for instance in the fabric-samples dir:
+
+```bash
+cd fabric-samples/token-sdk
+```
+
+> You can have the token-sdk folder anywhere. The commands keep working, as long as you have the TEST_NETWORK_HOME variable set.
+> See the bottom of this readme for instructions to use another Fabric network.
+
+### Quick start
+
+The quickest way to get going is to run:
+
+```bash
+./scripts/up.sh
+```
+
+This generates the crypto material, starts Fabric, deploys the chaincode, and starts the token nodes.
+
+When you're done and want to delete everything:
+
+```bash
+./scripts/down.sh
+```
+
+#### Using the application
+
+The services are running on the following ports:
+
+- 8080 API documentation
+- 9000 auditor
+- 9100 issuer
+- 9200 owner (alice and bob)
+- 9300 owner 2 (carlos and dan)
+
+Now let's issue and transfer some tokens! View the API documentation and try some actions at [http://localhost:8080](http://localhost:8080). Or, directly from the commandline:
+
+```bash
+curl -X POST http://localhost:9100/api/v1/issuer/issue -H 'Content-Type: application/json' -d '{
+    "amount": {"code": "EURX","value": 1000},
+    "counterparty": {"node": "owner1","account": "alice"},
+    "message": "hello world!"    
+}'
+
+curl -X GET http://localhost:9200/api/v1/owner/accounts
+
+curl -X POST http://localhost:9200/api/v1/owner/accounts/alice/transfer -H 'Content-Type: application/json' -d '{
+    "amount": {"code": "EURX","value": 100},
+    "counterparty": {"node": "owner1","account": "bob"},
+    "message": "hello bob!"    
+}'
+
+curl -X GET http://localhost:9200/api/v1/owner/accounts/bob/transactions
+```
```

Comment:
I think the largest missing piece is an explanation of what happened in each of the transactions.
- which nodes had to collaborate?
- what logic is in the FSC flow in the nodes versus in the chaincode on the Fabric peers?
- what data is stored on the nodes after each transaction?
- what data is stored on the Fabric ledger after each transaction?
- how is privacy maintained?
- what does the auditor do?